### PR TITLE
added api scan for book

### PR DESCRIPTION
**Pull request by mingyaang**
rip dip potato chip

### CODE REVIEW

File: app/src/main/java/com/team41/boromi/BookActivity.java [hunk 1]
```diff
@@ -328,4 +367,58 @@ public void onFailure(Exception e) {
...
+    System.out.println("DISPLAYING BOOKS");
+
+    GoogleBook gb = gson.fromJson(book, GoogleBook.class);
+    Book bookToDisplay = new Book();
+    bookToDisplay.setTitle(gb.getTitle(0));
+    bookToDisplay.setAuthor(gb.getFirstAuthor(0));
+    bookToDisplay.setISBN(isbn);
+    spinner.setVisibility(View.GONE);
+    DisplayBookFragment displayBookFragment = DisplayBookFragment.newInstance(bookToDisplay);
+    displayBookFragment.show(getSupportFragmentManager(), "displayUsers");
```
Comment (ravioli0509) [inline app/src/main/java/com/team41/boromi/BookActivity.java]: "displayBooks" ?

File: app/build.gradle [hunk 2]
```diff
@@ -66,4 +70,5 @@ dependencies {
     implementation 'com.google.firebase:firebase-auth:20.0.1'
     implementation 'com.google.firebase:firebase-messaging:21.0.0'
     implementation 'com.journeyapps:zxing-android-embedded:4.1.0'
+    implementation 'com.android.volley:volley:1.1.1'
```
Comment (ravioli0509) [inline app/build.gradle:73]: what does this implementation do

File: app/src/main/java/com/team41/boromi/book/DisplayBookFragment.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    authorTextView = (TextView) view.findViewById(R.id.available_author);
+    isbnTextView = (TextView) view.findViewById(R.id.available_isbn);
+
+    if (b.getTitle() == null)
+      titleTextView.setText("NO AVAILABLE TITLE");
+    else
+      titleTextView.setText(b.getTitle());
+
+    if(b.getAuthor() == null)
+      authorTextView.setText("NO AVAILABLE AUTHOR");
```
Comment (ravioli0509) [inline app/src/main/java/com/team41/boromi/book/DisplayBookFragment.java:71]: does the api also give a data for book image?

File: app/src/main/java/com/team41/boromi/BookActivity.java (hunk 1 above)
Comment (mingyaang) [reply app/src/main/java/com/team41/boromi/BookActivity.java]: lmfaooo I stole this code from your method, works either way XD but I will change it

File: app/build.gradle (hunk 2 above)
Comment (mingyaang) [reply app/build.gradle:73]: Allows for http requests through google's queue implementation

File: app/src/main/java/com/team41/boromi/book/DisplayBookFragment.java (hunk 3 above)
Comment (mingyaang) [reply app/src/main/java/com/team41/boromi/book/DisplayBookFragment.java:71]: It does, through the use of another api call of the image. But lowkey I am a little lazy to implement this

File: app/build.gradle (hunk 2 above)
Comment (ravioli0509) [reply app/build.gradle:73]: oh i see

File: app/src/main/java/com/team41/boromi/BookActivity.java (hunk 1 above)
Comment (ravioli0509) [reply app/src/main/java/com/team41/boromi/BookActivity.java]: lmfao you can leave it if it works

Comment (ravioli0509) [reply app/src/main/java/com/team41/boromi/BookActivity.java]: nvm u changed it

File: app/build.gradle (hunk 2 above)
Comment (mingyaang) [reply app/build.gradle:73]: https://developer.android.com/training/volley

File: app/src/main/java/com/team41/boromi/book/DisplayBookFragment.java (hunk 3 above)
Comment (ravioli0509) [reply app/src/main/java/com/team41/boromi/book/DisplayBookFragment.java:71]: oh thats tedious

**Review by ravioli0509 — APPROVED** [review]
LGTM, if we have time we can implement the book image bit as well but its not really necessary i suppose. Feel free to merge whenever you want to

**Pull request merged**